rossum_api/internal_sync_client.py

```diff
@@ -0,0 +1,286 @@
+            for bytes_chunk in self._stream(http_method, url, params=query_params):
+                yield bytes_chunk
+
+    def _stream(self, method: str, url: str, *args, **kwargs) -> Iterator[bytes]:
+        """Performs a streaming HTTP call."""
+        if not self.token:
+            self._authenticate()
+
+        # Do not force the calling site to alway prepend the base URL
+        enforce_domain(url, self.base_url)
+
+        for attempt in tenacity.Retrying(
+            wait=tenacity.wait_exponential_jitter(),
+            retry=tenacity.retry_if_exception(should_retry),
+            stop=tenacity.stop_after_attempt(self.n_retries),
+        ):
+            with (
+                attempt,
+                self.client.stream(
+                    method, url, headers=self._headers, *args, **kwargs
+                ) as response,
+            ):
+                if response.status_code == 401:
+                    self._authenticate()
+                    if attempt.retry_state.attempt_number == 1:
+                        raise AlwaysRetry()
+                self._raise_for_status(response)
+                for chunk in response.iter_bytes():
+                    yield chunk
+
+    def fetch_resource(
+        self, resource: Resource, id_: Union[int, str], request_params: dict[str, Any] = None
+    ) -> dict[str, Any]:
+        """Retrieve a single object in a specific resource.
+
+        Allows passing extra params specifically to allow disabling redirects feature of Tasks.
+        See https://elis.rossum.ai/api/docs/#task.
+        If redirects are desired, our raise_for_status wrapper must account for that.
+        """
+        return self.request_json("GET", f"{resource.value}/{id_}", params=request_params)
+
+    def fetch_resources(
+        self,
+        resource: Resource,
+        ordering: Sequence[str] = (),
+        sideloads: Sequence[str] = (),
+        content_schema_ids: Sequence[str] = (),
+        method: str = "GET",
+        json: Optional[dict] = None,
+        **filters,
+    ) -> Iterator[dict[str, Any]]:
+        """Retrieve a list of objects in a specific resource."""
+        for result in self.fetch_resources_by_url(
+            resource.value,
+            ordering,
+            sideloads,
+            content_schema_ids,
+            method,
+            json,
+            **filters,
+        ):
+            yield result
+
+    def fetch_resources_by_url(
+        self,
+        url: str,
+        ordering: Sequence[str] = (),
+        sideloads: Sequence[str] = (),
+        content_schema_ids: Sequence[str] = (),
+        method: str = "GET",
+        json: Optional[dict] = None,
+        **filters,
+    ) -> Iterator[dict[str, Any]]:
+        query_params = build_pagination_params(ordering)
+        query_params.update(build_sideload_params(sideloads, content_schema_ids))
+        query_params.update(**filters)
```

Comment:
No strong opinion but we could create a higher level function `build_fetch_resources_params(ordering, sideloads, content_schema_ids)`